### ISSUE TITLE
LGTM.com - false positive - This assignment to index is useless, since its value is never read.

### ISSUE THREAD

**Issue by sungam3r** [github/codeql#4895]
https://lgtm.com/projects/g/graphql-dotnet/graphql-dotnet/snapshot/49a3c43b84fac9156619f5ec61cd8e869bb28cdf/files/src/GraphQL/Utilities/StringUtils.cs?sort=name&dir=ASC&mode=heatmap#x420b05b499e825be:1

Its value used in `==` comparison.

**Comment by hvitved** [comment]
I agree that this is a false positive. (Note that rewriting `++index == itemsArray.Length - 1 ? "or " : ""` to `index == itemsArray.Length ? "or " : ""` does not work because it happens inside a lambda passed to `Aggregate`.)
